File: FinalBootMVC/src/main/java/com/springboot/pkg/model/Videoprojecteur.java
package com.springboot.pkg.model;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToOne;
import javax.persistence.Version;

@Entity
@DiscriminatorValue("V")
public class Videoprojecteur extends Materiel {
	
	private Module module;
	
	

	public Videoprojecteur() {
		super();
	}
	
	@OneToOne
	public Module getModule() {
		return module;
	}

	public void setModule(Module module) {
		this.module = module;
	}
	

}

File: FinalAngular/src/app/model/videoprojecteur.ts
import { Module } from './module';
import { Materiel } from './materiel';

export class VideoProjecteur extends Materiel {

    constructor(code?: number, cout?: number, private _module?: Module) {
        super(code, cout);
    }

    public get module(): Module {
        return this._module;
    }
    public set Module(value: Module) {
        this._module = value;
    }
}
